newIDE/app/src/Utils/Extension/ExtensionCompatibilityChecker.spec.js

```diff
@@ -0,0 +1,78 @@
+// @flow
+import {
+  getBreakingChanges,
+  formatExtensionsBreakingChanges,
+  formatOldBreakingChanges,
+  type ExtensionChange,
+} from './ExtensionCompatibilityChecker';
+import {
+  buttonV2ExtensionShortHeader,
+  breakingButtonV3ExtensionShortHeader,
+  breakingButtonV31ExtensionShortHeader,
+} from '../../fixtures/GDevelopServicesTestData';
+import { type ExtensionShortHeader } from '../GDevelopServices/Extension';
+
+describe('ExtensionCompatibilityChecker', () => {
+  describe('getBreakingChanges', () => {
+    it('can report empty breaking changes', () => {
+      expect(getBreakingChanges('1.0.0', buttonV2ExtensionShortHeader)).toEqual(
+        []
+      );
+    });
+
+    it('can report breaking changes', () => {
+      expect(
+        getBreakingChanges('1.0.0', breakingButtonV3ExtensionShortHeader)
+      ).toEqual([
+        {
+          version: '3.0.0',
+          changes:
+            '- The extension has breaking changes. It needs the following adaptations:\n  - first do this\n  - then this',
+        },
+      ]);
+    });
+
+    it('can exclude older breaking changes', () => {
+      expect(
+        getBreakingChanges('3.0.0', breakingButtonV31ExtensionShortHeader)
+      ).toEqual([]);
+    });
+  });
+
+  describe('formatExtensionsBreakingChanges', () => {
+    it('can format breaking changes', () => {
+      const breakingChanges = new Map<
+        ExtensionShortHeader,
+        Array<ExtensionChange>
+      >();
+      breakingChanges.set(breakingButtonV3ExtensionShortHeader, [
+        {
+          version: '3.0.0',
+          changes:
+            '- The extension has breaking changes. It needs the following adaptations:\n  - first do this\n  - then this',
+        },
+      ]);
+      expect(formatExtensionsBreakingChanges(breakingChanges)).toEqual(
```

Comment:
Note that you can use `toMatchInlineSnapshot` that may make these tests easier to write and maintain (because we can see the changes and update with a single key press if it's fine)